### PR TITLE
doc: release-notes-latest: Summarize mcuboot changes from the upmerge

### PR DESCRIPTION
**Pull request by nvlsianpu**
Summary on mcubooot, flash, usb, fcb, flash_map, settings,
stream_flash changes.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Review by jfischer-no — APPROVED** [review]

**Review by ru-fu — CHANGES_REQUESTED** [review]
Will add a commit with changes.

**Review by ru-fu — COMMENTED** [review]
Added a commit - please check and squash if okay. :)

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -105,3 +130,32 @@ The current |NCS| release is based on Zephyr 2.4.0.
 See the :ref:`Zephyr 2.4.0 release notes <zephyr:zephyr_2.4>` for a list of changes.
 
 For the list of the most recent additions specific to |NCS|, see :ref:`ncs_release_notes_140`.
+
+* Drivers and sensors:
+
+  * Flash:
+
+    * Added support for nRF53 Series SoCs in the nRF QSPI NOR flash driver (nrf_qspi_nor).
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:138]: This one has already been added - please remove.

**Review by ru-fu — CHANGES_REQUESTED** [review]
This PR is (in part?) duplicated in https://github.com/nrfconnect/sdk-nrf/pull/3484.
Please coordinate with @de-nordic .

**Comment by de-nordic** [comment]
@ru-fu Here https://github.com/nvlsianpu/fw-nrfconnect-nrf/pull/5 is PR to this PR that adds file systems and mcumgr.

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -94,8 +94,47 @@ MCUboot
 sdk-mcuboot
 -----------
 
-The MCUboot fork in |NCS| contains all commits from the upstream MCUboot repository up to and including ``5a6e18148d``, plus some |NCS| specific additions.
-The list of the most important recent changes can be found in :ref:`ncs_release_notes_140`.
```
Comment (nvlsianpu) [inline doc/nrf/releases/release-notes-latest.rst:98]: @ru-fu Should I restore this line?

Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:98]: No - we're listing changes since the last release, so we don't need that line in here anymore.

**Review by ru-fu — CHANGES_REQUESTED** [review]
Will add a commit.

**Comment by ioannisg** [comment]
Needs rebase @nvlsianpu

**Review by ru-fu — COMMENTED** [review]
Added - please check and squash, if okay.

File: doc/nrf/releases/release-notes-latest.rst [hunk 3]
```diff
@@ -331,8 +325,7 @@ The following list summarizes the most important changes inherited from upstream
   * File systems:
 
     * Enabled FCB to work with non-0xff erase value flash.
-    * Enabled FCB to work with non-0xff erase value flash.
-    * Added ``CONFIG_FS_MOUNT_FLAG_NO_FORMAT`` flag to FATFS Kconfig options, that removes formatting capabilities from FAT FS;
-      the option also prevents unformatted devices to be formatted to FAT FS on mount attempt.
-    * Added support for ``fs_mount()`` flags: ``FS_MOUNT_FLAG_READ_ONLY`` and ``FS_MOUNT_FLAG_NO_FORMAT``.
-    * The runtime check of FS API of a driver interface will not be perfromed when ``CONFIG_NO_RUNTIME_CHECKS`` option is enabled.
+    * Added a :c:macro:`FS_MOUNT_FLAG_NO_FORMAT` flag to the FatFs options.
+      This flag removes formatting capabilities from FatFs and prevents unformatted devices to be formatted to FatFs on mount attempt.
```
Comment (de-nordic) [inline doc/nrf/releases/release-notes-latest.rst:329]: The FatFs here refers FAT file system, where FAT means File Allocation Table this is not name of a library.

Comment (de-nordic) [reply doc/nrf/releases/release-notes-latest.rst:329]: Maybe, instead of `FatFs`, use `FAT file system`.

Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:329]: I'm not sure I understand where you refer to the library and where to the file system - please make a suggestion for the full entry.

Comment (de-nordic) [reply doc/nrf/releases/release-notes-latest.rst:329]: `This flag removes formatting capabilities from the FAT/exFAT file system driver and prevents unformatted devices to be formatted, to FAT or exFAT, on mount attempt.`

This is FAT https://en.wikipedia.org/wiki/File_Allocation_Table.
This is the library, FatFs, that supports it http://elm-chan.org/fsw/ff/00index_e.html.

The flags are generally for the FAT file system even though we use only one library for it, which is FatFs.
I see this way: we may use Groke named library, which would be supporting the FAT file system, but the user will still be formatting devices to FAT, or mounting FAT formatted devices; the user will not be trying to mount SD card formatted to Groke (that would be scary and would freeze the device anyway).

The devices available on the market, like SD cards, are usually formatted to FAT or exFAT.

Comment (nvlsianpu) [reply doc/nrf/releases/release-notes-latest.rst:329]: Subjesct of the first mention of `FatFS` is the library which implements FAT file system, the second mention mean the volume of FAT file system. 
Second mention should reflect that: `FAT file system' is more suitable there.

Comment (ru-fu) [reply doc/nrf/releases/release-notes-latest.rst:329]: OK, sounds good, please change as suggested. :)

File: doc/nrf/releases/release-notes-latest.rst [hunk 4]
```diff
@@ -306,3 +342,25 @@ The following list summarizes the most important changes inherited from upstream
...
+    * :ref:`flash_map_api`: Added an API to get the value of an erased byte in the flash_area.
+      See :c:func:`flash_area_erased_val`.
+    * :ref:`stream_flash`: Eliminated the usage of the flash API internals.
+
+
+  * File systems:
+
+    * Enabled FCB to work with non-0xff erase value flash.
+    * Added a :c:macro:`FS_MOUNT_FLAG_NO_FORMAT` flag to the FatFs options.
+      This flag removes formatting capabilities from FatFs and prevents unformatted devices to be formatted to FatFs on mount attempt.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:364]:
```suggestion
      This flag removes formatting capabilities from the FAT/exFAT file system driver and prevents unformatted devices to be formatted, to FAT or exFAT, on mount attempt.
```

**Review by ru-fu — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**